File: app/src/main/java/com/dam/authapp/MainActivity.java
package com.dam.authapp;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private EditText etUsuario;
    private EditText etPassword;
    private Button btnIngresar;
    private Button btnSalir;
    private SharedPreferences sharedPreferences;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Configurar barra de título
        if (getSupportActionBar() != null) {
            getSupportActionBar().setTitle(getString(R.string.login_title));
        }

        // Inicializar SharedPreferences
        sharedPreferences = getSharedPreferences("AuthAppPrefs", MODE_PRIVATE);

        // Vincular vistas
        etUsuario = findViewById(R.id.etUsuario);
        etPassword = findViewById(R.id.etPassword);
        btnIngresar = findViewById(R.id.btnIngresar);
        btnSalir = findViewById(R.id.btnSalir);

        // Validación de credenciales
        btnIngresar.setOnClickListener(v -> {
            String usuario = etUsuario.getText().toString().trim();
            String password = etPassword.getText().toString().trim();

            String savedUsuario = sharedPreferences.getString("username", null);
            String savedPassword = sharedPreferences.getString("password", null);

            if (usuario.equals(savedUsuario) && password.equals(savedPassword)) {
                // Credenciales correctas → navegar a HomeActivity
                Intent intent = new Intent(MainActivity.this, HomeActivity.class);
                startActivity(intent);
                finish();
            } else {
                // Credenciales incorrectas → mostrar error
                Toast.makeText(
                        MainActivity.this,
                        getString(R.string.error_credenciales),
                        Toast.LENGTH_LONG).show();
            }
        });

        // Cierre de la aplicación (botón)
        btnSalir.setOnClickListener(v -> finishAffinity());
    }

    // Inflar menú superior
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    // Manejar selección del menú
    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();

        // Navegación a Registro
        if (id == R.id.action_registrar) {
            Intent intent = new Intent(this, RegistrarActivity.class);
            startActivity(intent);
            return true;
        }

        // Cierre de la aplicación (menú)
        if (id == R.id.action_salir) {
            finishAffinity();
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/dam/authapp/MainActivity.kt
package com.dam.authapp

import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private lateinit var etUsuario: EditText
    private lateinit var etPassword: EditText
    private lateinit var btnIngresar: Button
    private lateinit var btnSalir: Button
    private lateinit var sharedPreferences: SharedPreferences

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Configurar barra de título
        supportActionBar?.title = getString(R.string.login_title)

        // Inicializar SharedPreferences
        sharedPreferences = getSharedPreferences("AuthAppPrefs", MODE_PRIVATE)

        // Vincular vistas
        etUsuario = findViewById(R.id.etUsuario)
        etPassword = findViewById(R.id.etPassword)
        btnIngresar = findViewById(R.id.btnIngresar)
        btnSalir = findViewById(R.id.btnSalir)

        // RF-02: Validación de credenciales
        btnIngresar.setOnClickListener {
            val usuario = etUsuario.text.toString().trim()
            val password = etPassword.text.toString().trim()

            val savedUsuario = sharedPreferences.getString("username", null)
            val savedPassword = sharedPreferences.getString("password", null)

            if (usuario == savedUsuario && password == savedPassword) {
                // Credenciales correctas → navegar a HomeActivity
                val intent = Intent(this, HomeActivity::class.java)
                startActivity(intent)
                finish()
            } else {
                // Credenciales incorrectas → mostrar error
                Toast.makeText(
                    this,
                    getString(R.string.error_credenciales),
                    Toast.LENGTH_LONG
                ).show()
            }
        }

        // RF-03: Cierre de la aplicación (botón)
        btnSalir.setOnClickListener {
            finishAffinity()
        }
    }

    // Inflar menú superior
    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    // Manejar selección del menú
    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            // RF-04: Navegación a Registro
            R.id.action_registrar -> {
                val intent = Intent(this, RegistrarActivity::class.java)
                startActivity(intent)
                true
            }
            // RF-03: Cierre de la aplicación (menú)
            R.id.action_salir -> {
                finishAffinity()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
}
